### ISSUE TITLE
Feature request: ability even not to sum up over all batches in loss functions

### ISSUE THREAD

**Issue by analvikingur** [pytorch/pytorch#1054]
Hi

One may want to evaluate perplexity in language modelling task. In this case it is necessary to average over sequence length, then take exponent and only then average over batch size.

For now it is impossible to implement such logic with `torch.nn.NLLLoss` in case of implicitly result averaging or summing up.

From my point of view, it may be reasonable not to perform any implicitly operations with result of loss functions to make them more flexible.

Cheers!

**Comment by kuangliu** [comment]
Maybe `size_average=False` should return a tensor of losses, instead of a number of `loss_sum`.

**Comment by apaszke** [comment]
Duplicate of #264
